pkgs/tools/networking/infiniband-diags/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, libtool, pkgconfig, rdma-core, glib, opensm }:
+
+stdenv.mkDerivation rec {
+  name = "infiniband-diags-${version}";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "linux-rdma";
+    repo = "infiniband-diags";
+    rev = version;
+    sha256 = "06x8yy3ly1vzraznc9r8pfsal9mjavxzhgrla3q2493j5jz0sx76";
+  };
+
+  nativeBuildInputs = [ autoconf automake libtool pkgconfig ];
+
+  buildInputs = [ rdma-core glib opensm ];
+
+  preConfigure = ''
+    export CFLAGS="-I${opensm}/include/infiniband"
+    ./autogen.sh
+  '';
+
```

Comment:
Thanks. FWIW, the approach in [perl-modules](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/perl-modules/generic/builder.sh) seems a little nicer, but it should ideally be factored out into a separate function like wrapProgram or patchShebangs. (Though I'm not sure how often it would be useful outside of actual perl packages...)